library/src/main/java/bio/terra/catalog/config/CatalogDatabaseConfiguration.java

```diff
@@ -43,4 +50,24 @@ public void setUpgradeOnStart(boolean upgradeOnStart) {
   public PlatformTransactionManager getTransactionManager() {
     return new JdbcTransactionManager(getDataSource());
   }
+
+  // This is a "magic bean": It supplies a method that Spring calls after the application is setup,
+  // but before the port is opened for business. That lets us do database migration and stairway
+  // initialization on a system that is otherwise fully configured. The rule of thumb is that all
+  // bean initialization should avoid database access. If there is additional database work to be
+  // done, it should happen inside this method.
+  @Bean
+  public SmartInitializingSingleton postSetupInitialization(ApplicationContext applicationContext) {
```

Comment:
Out of curiosity, why did this code end up changing from the BeanConfig? I just noticed the comment is the same

Reply:
It seemed like it belonged here more than in the other place, since it relies on the configuration in this object. I may have moved it around to try to get the database tests working too. I think it would work in either place; what I'd really like to do here is remove all the database and liquibase setup code and use the spring support instead. I'll check and make a ticket for that work if it doesn't exist already.